### PR TITLE
partition_manager: Include all regions in PM report

### PR DESCRIPTION
**Pull request by osaether**

### CODE REVIEW

File: scripts/partition_manager_report.py [hunk 1]
```diff
@@ -30,7 +30,11 @@ def print_region(domain, region, size, pm_config):
         bcolors = bcolors_ansi
 
     # Print header
-    print(bcolors["OKBLUE"] + "%s %s (0x%x - %dkB):" % (domain, region, size, size/1024) + bcolors["ENDC"])
+    # print(bcolors["OKBLUE"] + "%s %s (0x%x - %dkB):" % (region, size, size / 1024) + bcolors["ENDC"])
```
Comment (hakonfam) [inline scripts/partition_manager_report.py:33]: Should this be removed?

File: scripts/partition_manager_report.py [hunk 2]
```diff
@@ -30,7 +30,11 @@ def print_region(domain, region, size, pm_config):
...
+    if size >= 1024:
+        print(f"{bcolors['OKBLUE']} {domain} {region} (0x{size:x} - {int(size/1024):d}kB): {bcolors['ENDC']}")
```
Comment (hakonfam) [inline scripts/partition_manager_report.py:35]: Could you pull out the difference between these two prints and assign the value to a variable which you pass to a single print expression?

```
val = f'{int(size/1024):d}kB' if size >= 1024 else f'{size:d}B'
print(... val ...)
```

**Review by hakonfam — APPROVED** [review]

File: scripts/partition_manager_report.py [hunk 3]
```diff
@@ -90,12 +94,21 @@ def main():
...
-        max_address = max((part['address'] + part['size'] for part in pm_config_primary.values() if 'address' in part))
-        print_region(domain_name, max_address - min_address, pm_config_primary)
-
+            items = yaml.safe_load(f).items()
+        regions = []
+        for _, val in items:
+            r = val['region']
+            if r in regions:
+                continue
+            regions.append(r)
```
Comment (oyvindronningstad) [inline scripts/partition_manager_report.py:105]:
```suggestion
        regions = set(val['region'] for val in items.values())
        for r in regions:
```

File: scripts/partition_manager_report.py [hunk 4]
```diff
@@ -90,12 +94,21 @@ def main():
...
+            pm_config_primary = {k: v for k, v in items if v['region'] == r}
+            if len(pm_config_primary) == 0:
+                continue
```
Comment (oyvindronningstad) [inline scripts/partition_manager_report.py:108]: Curious: When would this happen?

File: scripts/partition_manager_report.py [hunk 5]
```diff
@@ -30,7 +30,8 @@ def print_region(region, size, pm_config):
         bcolors = bcolors_ansi
 
     # Print header
-    print(bcolors['OKBLUE'] + f'{region} ({hex(size)} - {size/1024}kB):' + bcolors['ENDC'])
+    val = f'{int(size/1024):d}kB' if size >= 1024 else f'{size:d}B'
+    print(f"{bcolors['OKBLUE']} {domain} {region} ({hex(size)} - {val}): {bcolors['ENDC']}")
```
Comment (oyvindronningstad) [inline scripts/partition_manager_report.py:34]:
```suggestion
    size_str = f'{int(size/1024):d}kB' if size >= 1024 else f'{size:d}B'
    print(f"{bcolors['OKBLUE']} {domain} {region} ({hex(size)} - {size_str}): {bcolors['ENDC']}")
```

Comment (oyvindronningstad) [reply scripts/partition_manager_report.py:34]: Maybe extract this logic into a function and use it below as well?

**Review by oyvindronningstad — COMMENTED** [review]
Some suggestions to make it more concise.

File: scripts/partition_manager_report.py [hunk 6]
```diff
@@ -41,11 +42,13 @@ def print_region(region, size, pm_config):
     sorted_pm_config = sorted(sorted_pm_config, key=lambda x: pm_config[x]['address'])
 
     # Create text lines
-    lines = ['{}{}: {} ({}){}'.format(
+    lines = ['{}{}: {} ({} - {}{}){}'.format(
                 '| '+bcolors['WARNING'] if 'span' not in pm_config[name] else '+---'+bcolors['OKBLUE'],
                 hex(pm_config[name]['address']),
                 name,
                 hex(pm_config[name]['size']),
+                pm_config[name]['size'] if pm_config[name]['size'] < 1024 else int(pm_config[name]['size']/1024),
```
Comment (tejlmand) [inline scripts/partition_manager_report.py:50]: If you use a lambda here, then you will signifficantly shorten the line:
```suggestion
               (lambda x: x if x < 1024 else int(x/1024))(pm_config[name]['size']),
```

**Review by tejlmand — CHANGES_REQUESTED** [review]

**Review by oyvindronningstad — APPROVED** [review]
LGTM!

**Review by osaether — COMMENTED** [review]
@tejlmand  Simplified by using a function.

File: scripts/partition_manager_report.py [hunk 7]
```diff
@@ -90,12 +94,21 @@ def main():
...
+            r = val['region']
+            if r in regions:
+                continue
+            regions.append(r)
+            pm_config_primary = {k: v for k, v in items if v['region'] == r}
+            if len(pm_config_primary) == 0:
+                continue
+            min_address = min((part['address'] for part in pm_config_primary.values() if 'address' in part))
+            max_address = max((part['address'] + part['size'] for part in pm_config_primary.values() if 'address' in part))
+            print_region(domain_name, r, max_address - min_address, pm_config_primary)
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply scripts/partition_manager_report.py:107]: well, you do add four additional spaces to each line, making too long lines even longer.
That corresponds to changing those lines in this PR :wink:

**Review by tejlmand — APPROVED** [review]

**Pull request merged**